src/FormElement/ScheduleElement/AnnuallyFields.php

```diff
@@ -0,0 +1,149 @@
+use ipl\Html\FormElement\FieldsetElement;
+use ipl\Html\HtmlElement;
+use ipl\Web\FormElement\ScheduleElement\Common\FieldsProtector;
+use ipl\Web\FormElement\ScheduleElement\Common\FieldsUtils;
+use ipl\Web\Widget\Icon;
+
+class AnnuallyFields extends FieldsetElement
+{
+    use FieldsUtils;
+    use FieldsProtector;
+
+    /** @var bool Whether the form is auto submitted */
+    protected $isAutoSubmitted = true;
+
+    /** @var array A list of valid months */
+    protected $months = [];
+
+    /** @var string A month to preselect by default */
+    protected $default = 'JAN';
+
+    public function __construct($name, $attributes = null)
+    {
+        $this->months = [
+            'JAN' => $this->translate('Jan'),
+            'FEB' => $this->translate('Feb'),
+            'MAR' => $this->translate('Mar'),
+            'APR' => $this->translate('Apr'),
+            'MAY' => $this->translate('May'),
+            'JUN' => $this->translate('Jun'),
+            'JUL' => $this->translate('Jul'),
+            'AUG' => $this->translate('Aug'),
+            'SEP' => $this->translate('Sep'),
+            'OCT' => $this->translate('Oct'),
+            'NOV' => $this->translate('Nov'),
+            'DEC' => $this->translate('Dec')
+        ];
+
+        parent::__construct($name, $attributes);
+    }
+
+    protected function init(): void
+    {
+        parent::init();
+        $this->initUtils();
+    }
+
+    public function onRegistered(Form $form)
+    {
+        $form->on(Form::ON_SENT, function ($form) {
+            $this->isAutoSubmitted = ! $form->hasBeenSubmitted();
+        });
+    }
+
+    /**
+     * Set the default month to be activated
+     *
+     * @param string $default
+     *
+     * @return $this
+     */
+    public function setDefault(string $default): self
+    {
+        if (! isset($this->months[strtoupper($this->default)])) {
+            throw new InvalidArgumentException(sprintf('Invalid month provided: %s', $default));
+        }
+
+        $this->default = strtoupper($default);
+
+        return $this;
+    }
+
+    protected function assemble()
+    {
+        $this->getAttributes()->set('id', $this->protectId('annually-fields'));
+
+        $fieldsSelector = new FieldsRadio('month', [
+            'class'     => 'autosubmit sr-only',
+            'value'     => $this->default,
+            'options'   => $this->months,
+            'protector' => function ($value) {
+                return $this->protectId($value);
+            }
+        ]);
+        $this->registerElement($fieldsSelector);
+
+        $runsOnThe = $this->getPopulatedValue('runsOnThe', 'n');
+        $this->addElement('checkbox', 'runsOnThe', [
+            'class' => 'autosubmit',
+            'value' => $runsOnThe
+        ]);
+
+        $checkboxControls = HtmlElement::create('div', ['class' => 'toggle-slider-controls']);
+        $checkbox = $this->getElement('runsOnThe');
+        $checkbox->prependWrapper($checkboxControls);
```

Comment:
I'm confident that you can avoid using wrappers here as well :wink:

Reply:
How? The checkbox is added using `addElement()`.

Reply:
Well, I missed that of course -.- now I'm not that confident anymore :rofl: